docs/source/architecture/events_and_transactions_receipts.rst

```diff
@@ -0,0 +1,343 @@
+transaction receipts: BlockEventExtractor and ReceiptEventExtractor. The
+BlockEventExtractor will extract events of type "block_commit" and
+"batch_commit". The ReceiptEventExtractor will extract events of type
+"state_delta" and events defined by transaction families.
+
+Example events generated by BlockEventExtractor:
+
+.. code-block:: protobuf
+
+  // Example block_commit event
+  Event {
+    event_type = "block_commit",
+    meta_data = [
+      Meta { key = "block_id", value = "abc...123" },
+      Meta { key = "block_num", value = "523" },
+      Meta { key = "state_root_hash", value = "def...456" },
+      Meta { key = "previous_block_id", value = "acf...146" },
+    ],
+  }
+
+  // Example block_commit event
+  Event {
+    event_type = "batch_commit",
+    meta_data = [Meta { key = "batch_id", value = "abc...123" }],
+  }
+
+Example events generated by ReceiptEventExtractor:
+
+.. code-block:: protobuf
+
+
+  // Example transaction family specific event
+  Event {
+    event_type = "xo_create",
+    meta_data = [Meta { key = "game_name", value = "game0" }],
+  }
+
+  // Example block_commit event
+  Event {
+    event_type = "state_delta",
+    meta_data = [Meta { key = "address", value = "abc...def" }],
+    event_data = <bytes>
+  }
+
+
+Transaction Receipts
+====================
+
+Transaction receipts support the following use cases:
+
+- Providing clients with information related to the execution of a transaction,
+  but which should not be stored in state, such as:
+
+    - Whether the transaction was valid
+    - How the transaction changed state
+    - Events of interest that occurred during execution of the transaction
+    - Other transaction family specific execution information
+
+- Providing the validator with information about transaction execution without
+  re-executing it.
+
+Sawtooth transaction receipts are represented as a protobuf message when exposed
+to clients. They contain a list of StateChange messages, a list of Event
+messages, and a list of TransactionReceipt.Data messages:
+
+.. code-block:: protobuf
+
+  message TransactionReceipt {
+    message Data {
+      // Interpretation left to transaction family
+      string data_type = 1;
+      bytes data = 2;
+    }
+    // State changes made by this transaction
+    // StateChange is already defined in protos/state_delta.proto
+    repeated StateChange state_changes = 1;
+    // Events fired by this transaction
+    repeated Event events = 2;
+    // Transaction family defined data
+    repeated Data data = 3;
+
+    string transaction_id = 4;
+  }
+
+The fields in the TransactionReceipt.Data are opaque to Sawtooth and their
+interpretation is left up to the transaction family. TransactionReceipt.Data
+can be requested for a transaction that has been committed and should only be
+used to store information about a transaction’s execution that should not be
+kept in state. Events produced by transaction processors can be subscribed to
```

Comment:
Suggested rewrite to avoid "can be subscribed to by":

     Clients can use the event system described above to subscribe to events produced by transaction processors.